pkg/scheduler/factory/factory.go

```diff
@@ -911,6 +893,24 @@ func (c *configFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 	}, nil
 }
 
+// getBinderFunc returns a func which returns an extender that supports bind or a default binder based on the given pod.
+func getBinderFunc(client clientset.Interface, extenders []algorithm.SchedulerExtender) func(pod *v1.Pod) Binder {
+	var extenderBinder algorithm.SchedulerExtender
+	for i := range extenders {
+		if extenders[i].IsBinder() {
+			extenderBinder = extenders[i]
+			break
+		}
+	}
+	defaultBinder := &binder{client}
```

Comment:
Can this be moved to before line 910 below (so that it is only called when needed) ?

Reply:
@tedyu thanks for the input. This PR has already merged, so any changes have to come in a new PR.

However, I think this line is in the right place. On line 906 we are creating a closure that captures the `defaultBinder` value, so this way we allocate once instead of every time we bind. (They are small objects, but I guess it's good to avoid all the extra allocations.) :smile:

Reply:
If the call to binder{client} is moved to between line 909 and line 910 , it would only be called when default binder is needed (not every time), right ?

Reply:
You are correct, but the main difference is that it would be called *each* time that the default binder is needed. `GetBinder` will be called for each pod that the scheduler binds.

Also, using the default binder is the common case.

Reply:
Thanks for the clarification.